src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/SqlProjectBuilderExtensions.cs

```diff
@@ -160,28 +169,51 @@ public static IResourceBuilder<SqlProjectResource> WithReference(
     public static IResourceBuilder<SqlPackageResource<TPackage>> WithReference<TPackage>(
         this IResourceBuilder<SqlPackageResource<TPackage>> builder, IResourceBuilder<SqlServerDatabaseResource> target)
         where TPackage : IPackageMetadata
+    {
+        return InternalWithReference(builder, target, target.Resource.DatabaseName);
+    }
+
+    /// <summary>
+    /// Publishes the SQL Server Database project to the target <see cref="IResourceWithConnectionString"/>.
+    /// </summary>
+    /// <param name="builder">An <see cref="IResourceBuilder{T}"/> representing the SQL Server Database project to publish.</param>
+    /// <param name="target">An <see cref="IResourceBuilder{T}"/> representing the target <see cref="IResourceWithConnectionString"/> to publish the SQL Server Database project to.</param>
+    /// <returns>An <see cref="IResourceBuilder{T}"/> that can be used to further customize the resource.</returns>
+    public static IResourceBuilder<SqlPackageResource<TPackage>> WithReference<TPackage>(
+        this IResourceBuilder<SqlPackageResource<TPackage>> builder,  IResourceBuilder<IResourceWithConnectionString> target)
+        where TPackage : IPackageMetadata
     {
         return InternalWithReference(builder, target);
     }
 
-    internal static IResourceBuilder<TResource> InternalWithReference<TResource>(this IResourceBuilder<TResource> builder, IResourceBuilder<SqlServerDatabaseResource> target)
+    internal static IResourceBuilder<TResource> InternalWithReference<TResource>(this IResourceBuilder<TResource> builder, IResourceBuilder<IResourceWithConnectionString> target, string? targetDatabaseName = null)
         where TResource : IResourceWithDacpac
     {
         builder.ApplicationBuilder.Services.TryAddSingleton<IDacpacDeployer, DacpacDeployer>();
         builder.ApplicationBuilder.Services.TryAddSingleton<SqlProjectPublishService>();
 
-        builder.ApplicationBuilder.Eventing.Subscribe<ResourceReadyEvent>(target.Resource, async (resourceReady, ct) =>
+        if (target.Resource is SqlServerDatabaseResource)
```

Comment:
This feels like a fair amount of duplicated code, with the only real difference being the event we subscribe to. Could we make that a bit cleaner? Perhaps extracting the event handler to a static method?